ocaml/libs/tracing/tracing.ml

```diff
@@ -624,6 +626,24 @@ module Tracer = struct
       let span = Span.start ~attributes ~name ~parent ~span_kind () in
       Spans.add_to_spans ~span ; Ok (Some span)
 
+  let update_span_with_parent span (parent : Span.t option) =
```

Comment:
as this is changing the underlying data structure, we probably want to guard it with the observe flag as in https://github.com/xapi-project/xen-api/blob/master/ocaml/libs/tracing/tracing.ml#L656 so thta it's a noop when tracing is not enabled

Reply:
https://github.com/xapi-project/xen-api/pull/5888/commits/28207c0632ba6492acf1bef49e30e964c864b76a